### DISCUSSION TITLE
Idefics2: how to send a post request with uploaded image?

### DISCUSSION THREAD

**Discussion by shizhi2** [huggingface/text-generation-inference#1911] [Q&A]
I know we can send the image via url in the form of 
`![](https://github.com/huggingface/text-generation-inference/assets/133914892/8174a6c4-3f46-4593-8293-b8bd76275eb4)`
How can we use a local image for the idefics2 post request? Looking for some examplea for both curl request and requests body in python. Thanks!

**Comment by ktrapeznikov** [comment]
https://huggingface.co/docs/text-generation-inference/basic_tutorials/visual_language_models

**Reply by shizhi2** [reply]
Thanks a lot! It seems that sending the base64 format image is only supported in python. How about sending the local image to cURL request?
